src/main/kotlin/com/curiouscreature/kotlin/math/Matrix.kt

```diff
@@ -386,7 +386,78 @@ fun transpose(m: Mat4) = Mat4(
         Float4(m.x.w, m.y.w, m.z.w, m.w.w)
 )
 fun inverse(m: Mat4): Mat4 {
-    TODO("Implement inverse(Mat4)") // TODO
+    val result = Mat4()
+    val pairs = FloatArray(12)
```

Comment:
The result matrix is only used at the end, it should be possible to skip the extra allocation of "pairs" and use only the result matrix. What do you think? It's not a big deal and the code is easier to read this way I suppose.

Reply:
I'm not sure if I understood you right, but I see two different approaches to eliminate extra allocation 
1) We can use stack variables
```
    val pair0  = m.z.z * m.w.w
    val pair1  =  m.z.z * m.w.w
    ...
    result.x.x  = pair0 * m.y.y + pair3 * m.z.y + pair4 * m.w.y
```
2) We can drop pairs precomputing
```
    result.x.x   =  m.z.z * m.w.w * m.y.y + m.y.z * m.w.w * m.z.y + m.w.z * m.y.w * m.w.y
    result.x.x -=  m.z.z * m.w.w * m.y.y + m.y.z * m.w.w * m.z.y + m.w.z * m.y.w * m.w.y
```

Reply:
Stack variables seem like the best approach, thanks!